pkg/perf/cache.go

```diff
@@ -0,0 +1,43 @@
+package perf
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/pkg/errors"
+)
+
+// TaskResult the result test schema
+type TaskResult struct {
+	TaskName     string      `json:"task_name"`
+	RunTimestamp string      `json:"run_timestamp"`
+	Result       interface{} `json:"result"`
+}
+
+// setCache set result in redis
+func (pm *PerformanceMonitor) setCache(ctx context.Context, taskName string, result TaskResult) error {
+	data, err := json.Marshal(result)
+	if err != nil {
+		return errors.Wrap(err, "failed to marshal data to JSON")
+	}
+
+	_, err = pm.redisConn.Do("SET", taskName, data)
+	return err
+}
+
+// GetCache get data from redis
+func (pm *PerformanceMonitor) GetCache(taskName string) (TaskResult, error) {
```

Comment:
As commented before, the GetCache is not correct. The performance framework internal details (how tests are run or how results are retrieved) is only an implementation details. The PerformanceMonitor struct then API should be `Get` for a result.

In other words, why a user of this framework know that the result is cached for example